### ISSUE TITLE
Вest way for running subscribers.

### ISSUE THREAD

**Issue by ssnickolay** [CJPoll/subscribex#9]
Hi, Cody!
I am a bit confused :)
When I configured `subscribex`, I didn't add it to my test env's applications stack (like it was written in README). So, when I launched tests, I saw "frozen" console...
It turned out that the reason in this line [this line](https://github.com/CJPoll/subscribex/blob/master/lib/subscribex.ex#L65 )
I added logs to this function and created [PR](https://github.com/CJPoll/subscribex/pull/8) about it.

But its not problem solution, and I wrote code similar to README:

```
defmodule MyApp do
  use Application

  def start(_type, _args) do
    import Supervisor.Spec

    children = [
      ...
    ] ++ subscribers(Mix.env)

    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
    Supervisor.start_link(children, opts)
  end

  defp subscribers(:test), do: []
  defp subscribers(_) do
    import Supervisor.Spec
    [ worker(MySubscriber, []) ]
  end
end

```
What do you think about it? It is probably worth adding a certain flag `test_mode` and implementing different logic inside library?
Because also I don't want to install RabbitMQ to my CI container...

p/s/ and thx for this library 👍

**Comment by CJPoll** [comment]
I agree, having a test mode in config to _not_ start subscribex would be much better than what I've been doing. Thanks for the idea! I'll see if I can get that out in the next day or so for you.

Here's how I've gone about it:

```elixir
def application do
    apps = [
        # Dependencies to start up other than subscribex
    ]

    do_application(apps, Mix.env)
  end

  def do_application(apps, :test) do
    [mod: {MyApp, []}, applications: apps]
  end

  def do_application(apps, _env) do
    [mod: {MyApp, []}, applications: apps ++ [:subscribex]]
  end
```

Also, you're more than welcome :)
